app/models/nic/base.rb

```diff
@@ -346,7 +346,7 @@ def validate_mac_is_unicast
     end
 
     def validate_updating_types
-      sti_type = type || 'Nic::Base'
+      sti_type = type
       errors.add(:type, _("can't be changed once the interface is saved")) if persisted? && (self.class.name != sti_type)
```

Comment:
Isn't this the same as:

```suggestion
      errors.add(:type, _("can't be changed once the interface is saved")) if persisted? && (self.class.name != type)
```

Reply:
It is, truth be told I don't know why I left it like this. Applied.